pkg/operator/controllers/routefix/routefix.go

```diff
@@ -39,96 +35,25 @@ do
 done`
 )
 
-type RouteFix interface {
-	CreateOrUpdate(ctx context.Context) error
-}
-
-type routeFix struct {
-	log *logrus.Entry
-	env env.Interface
-
-	cli    kubernetes.Interface
-	seccli securityclient.Interface
-}
-
-func New(log *logrus.Entry, e env.Interface, cli kubernetes.Interface, seccli securityclient.Interface) RouteFix {
-	return &routeFix{
-		log: log,
-		env: e,
-
-		cli:    cli,
-		seccli: seccli,
-	}
-}
-
-func (i *routeFix) routefixImage() string {
-	return fmt.Sprintf(routefixImageFormat, i.env.ACRName())
-}
-
-func (i *routeFix) ensureNamespace(ctx context.Context, ns string) error {
-	_, err := i.cli.CoreV1().Namespaces().Create(ctx, &v1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:        ns,
-			Annotations: map[string]string{projectv1.ProjectNodeSelector: ""},
-		},
-	}, metav1.CreateOptions{})
-	if !errors.IsAlreadyExists(err) {
-		return err
-	}
-
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		_ns, err := i.cli.CoreV1().Namespaces().Get(ctx, ns, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		if _ns.Annotations == nil {
-			_ns.Annotations = map[string]string{}
-		}
-
-		if annotation, ok := _ns.Annotations[projectv1.ProjectNodeSelector]; annotation == "" && ok {
-			return nil
-		}
-
-		_ns.Annotations[projectv1.ProjectNodeSelector] = ""
-
-		_, err = i.cli.CoreV1().Namespaces().Update(ctx, _ns, metav1.UpdateOptions{})
-		return err
-	})
+func (r *RouteFixReconciler) routefixImage() string {
+	return fmt.Sprintf(routefixImageFormat, "arosvc")
 }
 
-func (i *routeFix) applyDaemonSet(ctx context.Context, ds *appsv1.DaemonSet) error {
-	_, err := i.cli.AppsV1().DaemonSets(ds.Namespace).Create(ctx, ds, metav1.CreateOptions{})
-	if !errors.IsAlreadyExists(err) {
-		return err
-	}
-
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		_ds, err := i.cli.AppsV1().DaemonSets(ds.Namespace).Get(ctx, ds.Name, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		ds.ResourceVersion = _ds.ResourceVersion
-		_, err = i.cli.AppsV1().DaemonSets(ds.Namespace).Update(ctx, ds, metav1.UpdateOptions{})
-		return err
-	})
-}
-
-func (i *routeFix) CreateOrUpdate(ctx context.Context) error {
-	err := i.ensureNamespace(ctx, kubeNamespace)
+func (r *RouteFixReconciler) Ensure(ctx context.Context) error {
```

Comment:
All of this code is so similar to the genevalogging controller.  You're better off copying that, including:
* its use of the dynamic helper to handle ensure/create/update logic
* its reconcile pattern (requeueing of objects to the owning cluster object)